datura/s3_minio_test.go

```diff
@@ -0,0 +1,203 @@
+	viper.SetConfigType("yml")
+	viper.SetConfigName(".s3_minio_test")
+	viper.AutomaticEnv()
+
+	errnie.Handles(
+		viper.ReadInConfig(),
+	)
+}
+
+/*
+Setup an ebedded local Minio instance
+*/
+func SetupMinio(name string, t *testing.T) (string, func() error, error) {
+	dir := t.TempDir()
+	go minio.Main([]string{"minio", "server", "--quiet", "--address", addr, dir})
+	// TODO replace with Readiness check
+	time.Sleep(2 * time.Second)
+
+	madm, err := madmin.New(addr, MINIOKEY, MINIOSECRET, false)
+	if err != nil {
+		return "", nil, errnie.NewError(
+			fmt.Errorf("while creating madmin.. %s", err),
+		)
+	}
+
+	// Fetch service status.
+	_, err = madm.ServerInfo(context.TODO())
+	if err != nil {
+		return "", nil, errnie.NewError(
+			fmt.Errorf("while getting madmin info.. %s", err),
+		)
+	}
+
+	mc, err := mclient.New(addr, &mclient.Options{
+		Creds:  credentials.NewStaticV4(MINIOKEY, MINIOSECRET, ""),
+		Secure: false,
+	})
+
+	errnie.Handles(
+		mc.MakeBucket(context.Background(), name, mclient.MakeBucketOptions{}),
+	)
+
+	return addr, func() error {
+		t.Cleanup(func() {
+			errnie.Handles(
+				os.RemoveAll(dir),
+			)
+			errnie.Handles(
+				madm.ServiceStop(context.Background()),
+			)
+		})
+
+		return nil
+	}, nil
+
+}
+
+/*
+Just a preliminary connection attempt to the embedded minio server. Does not really
+serve a testing purpose as we are using the AWS client rather than the Minio client
+*/
+func TestMinioPut(t *testing.T) {
+	var cleanup func() error
+	var err error
+	addr, cleanup, err = SetupMinio("test", t)
+	if err != nil {
+		t.Fatalf("while starting embedded server: %s\n", err)
+	}
+
+	var mc *mclient.Client
+	Convey("Given a new minio client", t, func() {
+		mc, err = mclient.New(addr, &mclient.Options{
+			Creds:  credentials.NewStaticV4(MINIOKEY, MINIOSECRET, ""),
+			Secure: false,
+		})
+
+		Convey("There should be no error", func() {
+			So(
+				err,
+				ShouldBeNil,
+			)
+		})
+	})
+
+	data := []byte("test")
+	Convey("Using the minio client", t, func() {
+		_, err = mc.PutObject(
+			context.Background(),
+			"test",
+			"foo/var",
+			bytes.NewReader(data),
+			int64(len(data)),
+			mclient.PutObjectOptions{},
+		)
+
+		Convey("Writing an object should give no error", func() {
+			So(
+				err,
+				ShouldBeNil,
+			)
+		})
+	})
+
+	errnie.Handles(
+		cleanup(),
+	)
+}
+
+/*
+Write a datagram to the embedded Minio bucket and count the objects in the bucket.
+*/
+func TestDaturaS3WriteDatagram(t *testing.T) {
+	var cleanup func() error
+	var err error
+	addr, cleanup, err = SetupMinio("cnc-development-datalake", t)
+	if err != nil {
+		t.Fatalf("while starting embedded server: %s\n", err)
+	}
+
+	// sets the addr of the Minio endpoint
+	initConfig(t)
+
+	mc := datura.NewS3()
+	t.Logf("S3 Client Bucket: %s", *mc.Bucket)
+	t.Logf("S3 Client Client: %+v", *mc.Client)
+	t.Logf("S3 Client Region: %s", mc.Region)
+	t.Logf("S3 Client Ctx: %+v", *mc.Ctx)
+	t.Logf("S3 Client Pool: %+v", *mc.Pool)
+
+	// create a Pool
```

Comment:
Should the test not be just this? https://github.com/TheApeMachine/wrkspc/blob/master/datura/s3_test.go?ts=4
I just copied it from the radix test, since they are both the same interface.
I have added a test environment here: https://github.com/TheApeMachine/wrkspc/blob/master/cmd/cfg/.wrkspc.yml?ts=4 (though left stage on development) and building the client etc. is just done by the actual package, not by the test.

Reply:
Yes, that's it!

Reply:
where is this function?

Reply:
[https://github.com/TheApeMachine/wrkspc/blob/7fa84a38979e08bd11262044913e2645ea11c85b/datura/s3_minio_test.go#L64](https://github.com/TheApeMachine/wrkspc/blob/7fa84a38979e08bd11262044913e2645ea11c85b/datura/s3_minio_test.go#L64)